framework/src/endpoint.ts

```diff
@@ -21,12 +21,25 @@ export const isReservedEndpointFunction = (key: string): boolean =>
 
 export const getEndpointHandlers = (endpoint: Record<string, unknown>): EndpointHandlers => {
 	const endpointHandlers: EndpointHandlers = {};
-	for (const key of Object.getOwnPropertyNames(Object.getPrototypeOf(endpoint))) {
-		const val = endpoint[key];
-		if (!isReservedEndpointFunction(key) && typeof val === 'function') {
-			endpointHandlers[key] = val.bind(endpoint) as EndpointHandler;
+	let isBaseEndpoint;
+
+	// For endpoints which extend other endpoints, go deeper in the prototype chain
+	do {
+		// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, no-param-reassign
+		endpoint = Object.getPrototypeOf(endpoint);
+
+		for (const key of Object.getOwnPropertyNames(endpoint)) {
+			const val = endpoint[key];
+			if (!isReservedEndpointFunction(key) && typeof val === 'function') {
+				endpointHandlers[key] = val.bind(endpoint) as EndpointHandler;
+			}
 		}
-	}
+
+		isBaseEndpoint =
+			(Object.getPrototypeOf(Object.getPrototypeOf(endpoint)) as Record<string, unknown>)
+				.constructor.name === Object.name;
```

Comment:
why is this comparing with `Object.name`('Object')?

Reply:
What would be a more elegant way to detect that the loop has reached the end where the prototype returned by `Object.getPrototypeOf()` is `Object`?